sources/packages/api/src/route-controllers/report/report.aest.controller.ts

```diff
@@ -0,0 +1,144 @@
+  Body,
+  Controller,
+  Post,
+  Res,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+import {
+  ApiBadRequestResponse,
+  ApiTags,
+  ApiUnprocessableEntityResponse,
+} from "@nestjs/swagger";
+import { Response } from "express";
+import { Readable } from "stream";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
+import { UserGroups } from "../../auth/user-groups.enum";
+import { ReportService, FormService } from "../../services";
+import { ClientTypeBaseRoute } from "../../types";
+import { StringBuilder } from "../../utilities/string-builder";
+import {
+  getDateOnlyFormatAsString,
+  getFileNameTimestamp,
+  CustomNamedError,
+} from "../../utilities";
+import BaseController from "../BaseController";
+import { ReportsFilterAPIInDTO } from "./models/report.dto";
+import { FormNames } from "../../services/form/constants";
+import {
+  REPORT_CONFIG_NOT_FOUND,
+  FILTER_PARAMS_MISMATCH,
+} from "../../services/report/constants";
+
+/**
+ * Controller for Reports for AEST Client.
+ * This consists of all Rest APIs for notes.
+ */
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("reports")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-reports`)
+export class ReportAESTController extends BaseController {
+  constructor(
+    private readonly reportService: ReportService,
+    private readonly formService: FormService,
+  ) {
+    super();
+  }
+
+  /**
+   * Rest API to financial reports in csv format.
+   * @param payload report filter payload.
+   * @param response
+   */
+  @ApiBadRequestResponse({
+    description: "Not able to export report due to an invalid request.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Either report config missing or filter parameters not matching the report config.",
+  })
+  @Post()
+  async exportReport(
+    @Body() payload: ReportsFilterAPIInDTO,
+    @Res() response: Response,
+  ): Promise<void> {
+    const submissionResult = await this.formService.dryRunSubmission(
+      FormNames.ExportFinancialReports,
+      payload,
+    );
+
+    if (!submissionResult.valid) {
+      throw new BadRequestException(
+        "Not able to export report due to an invalid request.",
+      );
+    }
+
+    let reportData = [];
+
+    try {
+      reportData = await this.reportService.exportReport(payload);
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case REPORT_CONFIG_NOT_FOUND:
+          case FILTER_PARAMS_MISMATCH:
+            throw new UnprocessableEntityException(error.message);
+          default:
+            throw error;
+        }
+      }
+      throw error;
+    }
+
+    if (!reportData || reportData.length === 0) {
+      this.streamFile(response, payload.reportName, "No data found.");
+      return;
+    }
+
+    const reportCSVContent = new StringBuilder();
```

Comment:
can you add inline comments for this file logic, it will be helpful for other developers.